src/aiida_workgraph/orm/workgraph.py

```diff
@@ -11,28 +11,167 @@
+        """
+        Return the task process info
+
+        :returns: string representation of the task process info
+        """
+        return self.base.attributes.get(self.TASK_PROCESSES_KEY, {})
+
+    def set_task_processes(self, task_processes: str) -> None:
+        """
+        Set the task process info
+
+        :param process: string representation of the task process info
+        """
+        return self.base.attributes.set(self.TASK_PROCESSES_KEY, task_processes)
+
+    def get_task_process(self, task_name: str) -> Optional[str]:
+        """
+        Return the task state info
+
+        :returns: string representation of the task state info
+        """
+        return self.task_processes.get(task_name, None)
+
+    def set_task_process(self, task_name: str, task_process: str) -> Optional[str]:
+        """
+        Set the task state info
+
+        :returns: string representation of the task state info
+        """
+        task_processes = self.task_processes
+        task_processes[task_name] = task_process
+        return self.set_task_processes(task_processes)
+
+    @property
+    def task_actions(self) -> Optional[str]:
+        """
+        Return the task action info
+
+        :returns: string representation of the task action info
+        """
+        return self.base.attributes.get(self.TASK_ACTIONS_KEY, {})
+
+    def set_task_actions(self, task_actions: str) -> None:
+        """
+        Set the task action info
+
+        :param action: string representation of the task action info
+        """
+        return self.base.attributes.set(self.TASK_ACTIONS_KEY, task_actions)
+
+    def get_task_action(self, task_name: str) -> Optional[str]:
+        """
+        Return the task action info
+
+        :returns: string representation of the task action info
+        """
+        return self.task_actions.get(task_name, "")
+
+    def set_task_action(self, task_name: str, task_action: str) -> Optional[str]:
+        """
+        Set the task action info
+
+        :returns: string representation of the task action info
+        """
+        task_actions = self.task_actions
+        task_actions[task_name] = task_action
+        return self.set_task_actions(task_actions)
+
+    @property
+    def workgraph_data(self) -> Optional[str]:
+        """
+        Return the workgraph data
+
+        :returns: string representation of the workgraph data
+        """
+        return self.base.attributes.get(self.WORKGRAPH_DATA_KEY, None)
+
+    def set_workgraph_data(self, workgraph_data: dict) -> None:
+        """
+        Set the workgraph data
+
+        :param data: string representation of the workgraph data
+        """
+        return self.base.attributes.set(self.WORKGRAPH_DATA_KEY, workgraph_data)
 
     @property
-    def workgraph_state_info(self) -> Optional[str]:
+    def workgraph_data_short(self) -> Optional[str]:
         """
-        Return the workgraph state info
+        Return the workgraph data
```

Comment:
Can you give more explanation what `data_short` is in comparison to just `data`?

Reply:
The workgraph data is the data that only has minimal information to be displayed in the GUI. I update the docstring.